File: app/src/main/java/io/treehouses/remote/bases/BaseFragment.java
package io.treehouses.remote.bases;

import android.bluetooth.BluetoothAdapter;
import android.content.Context;
import android.content.Intent;
import android.os.Handler;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import io.treehouses.remote.Constants;
import io.treehouses.remote.Network.BluetoothChatService;
import io.treehouses.remote.callback.HomeInteractListener;

public class BaseFragment extends Fragment {
    public BluetoothChatService mChatService;

    public HomeInteractListener listener;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (context instanceof HomeInteractListener)
            listener = (HomeInteractListener) context;
        else
            throw new RuntimeException("Implement interface first");

        mChatService = listener.getChatService();

    }

    public void setupChat() {

    }

    protected void onLoad(Handler mHandler) {
        mChatService = listener.getChatService();
        mChatService.updateHandler(mHandler);
        // If the adapter is null, then Bluetooth is not supported
        if (!mChatService.isBluetoothSupported()) {
            Toast.makeText(getActivity(), "Bluetooth is not available", Toast.LENGTH_LONG).show();
            getActivity().finish();
        }
        if (!mChatService.isBluetoothEnabled()) {
            Intent enableIntent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
            startActivityForResult(enableIntent, Constants.REQUEST_ENABLE_BT);
        }
    }


}

File: app/src/main/java/io/treehouses/remote/bases/BaseFragment.kt
package io.treehouses.remote.bases

import android.bluetooth.BluetoothAdapter
import android.content.Context
import android.content.Intent
import android.os.Handler
import android.widget.Toast
import androidx.fragment.app.Fragment
import io.treehouses.remote.Constants
import io.treehouses.remote.Network.BluetoothChatService
import io.treehouses.remote.callback.HomeInteractListener

open class BaseFragment : Fragment() {
    lateinit var mChatService: BluetoothChatService
    var mBluetoothAdapter: BluetoothAdapter? = null
    lateinit var listener: HomeInteractListener
    override fun onAttach(context: Context) {
        super.onAttach(context)
        listener = if (context is HomeInteractListener) context else throw RuntimeException("Implement interface first")
    }

    protected fun onLoad(mHandler: Handler?) {
        mChatService = listener.getChatService()
        mChatService.updateHandler(mHandler!!)
        mBluetoothAdapter = BluetoothAdapter.getDefaultAdapter()

        // If the adapter is null, then Bluetooth is not supported
        if (mBluetoothAdapter == null) {
            Toast.makeText(activity, "Bluetooth is not available", Toast.LENGTH_LONG).show()
            requireActivity().finish()
        }
        checkStatusNow()
        if (!mBluetoothAdapter!!.isEnabled) {
            val enableIntent = Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE)
            startActivityForResult(enableIntent, Constants.REQUEST_ENABLE_BT)
        } else {
            setupChat()
        }
    }

    fun checkStatusNow() {}
    open fun setupChat() {}
}